sense-android-library/src/nl/sense_os/service/storage/DbHelper.java

```diff
@@ -67,4 +121,35 @@ public void onUpgrade(SQLiteDatabase db, int oldVers, int newVers) {
         db.execSQL("DROP TABLE IF EXISTS " + DbHelper.TABLE);
         onCreate(db);
     }
-}
\ No newline at end of file
+
+    public SQLiteDatabase getWritableDatabase(){
+      return getWritableDatabase(passphrase);
+    }
+
+    public SQLiteDatabase getReadableDatabase(){
+      return getReadableDatabase(passphrase);
+    }
+
+  private void setPassphrase(String imei) {
+
+    byte[] passphrase_string = imei.getBytes();
+    byte[] passphrase_salt = PASSPHRASE_SALT.getBytes();
+
+    byte[] input = new byte[passphrase_string.length + passphrase_salt.length];
+    System.arraycopy(passphrase_string, 0, input, 0, passphrase_string.length);
+    System.arraycopy(passphrase_salt, 0, input, passphrase_string.length, passphrase_salt.length);
+
+    MessageDigest md = null;
+    try {
+      md = MessageDigest.getInstance("SHA-1");
+    } catch (NoSuchAlgorithmException e1) {
+      Log.e(TAG, "Error initializing SHA1 message digest");
+    }
+
+    byte[] result_byte = md.digest(input);
```

Comment:
@orhan89 I guess md.digest((imei+PASSPHRASE_SALT).getBytes()) also works?
And is SHA-1 good enough or do we need a stronger encryption?